src/com/itacademy/eshop/shop/Eshop.java

```diff
@@ -25,16 +26,22 @@ public void printProducts() {
         }
     }
 
-    public void addProduct(Product shirt) {
-        products.add(shirt);
-    }
+    public void addProduct(Product shirt) throws DuplicateProductException {
```

Comment:
i dont think this method works the way you want it to.
in the if statement, you are saying that if new item matches an existing product, it should be added to products list, otherwise, throw duplicate exception. so if you have an empty list here, you would get a DuplicateProductException